packages/internal/src/generate/templates/all-currentUser.d.ts.template

```diff
@@ -6,9 +6,20 @@ import { getCurrentUser } from '../../../api/src/lib/auth'
 type ThenArg<T> = T extends PromiseLike<infer U> ? U : T
 export type InferredCurrentUser = ThenArg<ReturnType<typeof getCurrentUser>>
 
+type UndefinedRoles = {
+  /**
+   * ⚠️ Heads-Up: This is undefined unless roles key is returned from {@link getCurrentUser()}
+   *
+   * You may look into to https://redwoodjs.com/docs/tutorial/chapter7/rbac for a tutorial on
+   * how to add fully-fledged RBAC (Role-based Access Control) to your database model.
+   */
+  roles?
+}
+
+type Overwrite<T, U> = Pick<T, Exclude<keyof T, keyof U>> & U;
```

Comment:
I think your `Overwrite` type can be simplified like this:
```suggestion
type Overwrite<T, U> = Omit<T, keyof U> & U;
```

Reply:
:clap: you're right, and that one has much less cognitive complexity, i like!

Reply:
Resolved via 539844f519477869c417cc0988c2e886fe3a3ea5